### PR TITLE
Added unit testing, snapshot testing, and one server integration test.

### PR DESCRIPTION
**Pull request by jmbice**
Had to configure my yml and jest files to get it to work in circleci

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -29,12 +31,18 @@ jobs:
           - v1-dependencies-
 
       - run: yarn install
+      - run: sudo apt install -y mysql-client
+      # - run: sudo apt-get install mysql-server
```
Comment (burkegg) [inline .circleci/config.yml]: Are you leaving this code in as a future option?

File: client/components/Appointment.jsx [hunk 2]
```diff
@@ -15,6 +15,10 @@ const outerShell = {
 class Appointment extends React.Component {
   constructor(props) {
     super(props);
+    this.next = this.next.bind(this);
+    this.back = this.back.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.makeAppointment = this.makeAppointment.bind(this);
     this.state = {
       businessId: Math.floor(Math.random() * Math.floor(100)),
```
Comment (burkegg) [inline client/components/Appointment.jsx]: If you're going for a number between 0 and 99 inclusive, I think you can leave off the second Math.floor.
i.e.  Math.floor(Math.random() * 100)

File: db/loadFake.js [hunk 3]
```diff
@@ -1,12 +1,11 @@
...
-const hoursOpen = [6, 7, 8, 9, 10, 11, 12];
-const afterHours = ['01', '02'];
-const minutes = ['00', '30'];
-
 const generateFakeBusinessData = () => {
+  const openingTimes = ['06', '07', '08', '09', '10', '11', '12', '13', '14', '15', '16', '17', '18'];
+  const hoursOpen = [6, 7, 8, 9, 10, 11, 12];
+  const afterHours = ['01', '02'];
+  const minutes = ['00', '30'];
   const getRandom = max => Math.floor(Math.random() * Math.floor(max));
```
Comment (burkegg) [inline db/loadFake.js]: Unless you're putting in a decimal for max, I don't think Math.floor(max) is going to do anything but give you max.  So it definitely doesn't hurt anything to have it there, but I don't think it's doing anything.

File: tests/Appointment.test.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+  it('makeAppointment should call forceUpdate', (done) => {
+    wrapper.instance().forceUpdate = jest.fn();
+    wrapper.instance().makeAppointment();
+    done();
+    expect(wrapper.instance().forceUpdate).toHaveBeenCalledTimes(1);
+  });
+});
+
+
+// it('makeAppointment should make as forceUpdate', (done) => {
```
Comment (burkegg) [inline tests/Appointment.test.js]: commented code.

File: tests/LevelOne.test.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+          count={'2'}
+          date={'July 26, 2018'}
+          time={'9:30 AM'}
+        />
+      );
+
+      expect(toJson(levelOneComponent)).toMatchSnapshot();
+    });
+
+    it('getGuestCount should return an array of numbers, or null', () => {
```
Comment (burkegg) [inline tests/LevelOne.test.js:29]: This strikes me as a really nice test for both positive and negative results.

File: tests/LevelOne.test.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  });
+
+  it('should render 3 select options and a button', () => {
+    expect(wrapper.find('select#selectedDate').exists()).toEqual(true);
+    expect(wrapper.find('select#selectedTime').exists()).toEqual(true);
+    expect(wrapper.find('select#guestCount').exists()).toEqual(true);
+    expect(wrapper.find('button').exists()).toEqual(true);
+  });
+});
+//
```
Comment (burkegg) [inline tests/LevelOne.test.js]: commented code.

**Review by burkegg — CHANGES_REQUESTED** [review]
There is some commented code, and two places where you have a tiny bit of code that doesn't hurt, but doesn't do anything either.  Your tests look great.  Nice work!

File: .circleci/config.yml (hunk 1 above)
Comment (jmbice) [reply .circleci/config.yml]: Not sure why I left that in there...

File: client/components/Appointment.jsx (hunk 2 above)
Comment (jmbice) [reply client/components/Appointment.jsx]: I did not know this!

File: db/loadFake.js (hunk 3 above)
Comment (jmbice) [reply db/loadFake.js]: I didn't have a firm understanding how this worked, this is most helpful.

File: tests/LevelOne.test.js (hunk 6 above)
Comment (jmbice) [reply tests/LevelOne.test.js]: On it!

File: tests/LevelOne.test.js (hunk 5 above)
Comment (jmbice) [reply tests/LevelOne.test.js:29]: You're too kind sir.

File: .circleci/config.yml (hunk 1 above)
Comment (jmbice) [reply .circleci/config.yml]: I am going to try it without, see if it's true. If not, i'll omit it on my next PR.

**Pull request merged**